app/scripts/directives/mobileClientBuildsList.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+(function() {
+  angular.module('openshiftConsole').component('mobileClientBuildsList', {
+    controller: [
+      '$timeout',
+      '$anchorScroll',
+      '$location',
+      'APIService',
+      'DataService',
+      MobileClientBuildsList
+    ],
+    bindings: {
+      mobileClient: '<'
+    },
+    templateUrl: 'views/mobile-client-builds-list.html'
+  });
+
+  function MobileClientBuildsList(
+    $timeout,
+    $anchorScroll,
+    $location,
+    APIService,
+    DataService
+  ) {
+    var ctrl = this;
+    var buildConfigsVersion = APIService.getPreferredVersion('buildconfigs');
+    var watches = [];
+    var anchor;
+    var PAGE_HEADER_HEIGHT = 100;
+
+    ctrl.$onChanges = function(changes) {
+      var mobileClientChanged = _.get(changes, 'mobileClient.currentValue', false);
+      if (mobileClientChanged && !ctrl.context) {
+        ctrl.context = {namespace: _.get(ctrl, 'mobileClient.metadata.namespace')};
+      }
+
+      if (mobileClientChanged && !ctrl.buildsWatchSet) {
```

Comment:
It may be helpful to introduce some named functions here to self-document.  I want to know why `$onChanges` could trigger `goToBuildConfig()`.